File: revision/max_consecutive_ones.cpp
#include<iostream>
#include<vector>
using namespace std;
void input(vector<int>&nums){
   for(auto &i : nums){
       cin >> i;
   }
}
void display(vector<int>nums){
    for(auto const i : nums){
        cout << i << ' ' ;
    }
    cout << endl;
}
int findMaxConsecutiveOnes(vector<int>&nums){
    //nums = [1,1,0,1,1,1], output : 3
    //Linear Search O(N)
    int check , max;
    max = check = 0;
    for(int i = 0;i < nums.size(); i++){
        if (nums[i] == 1){
            check++;
            if (check >= max){
                max = check;
            }
        } else {
            check = 0;
        }
    }
    return max;
}
int main()
{
    int n;
    cin >> n;
    vector<int>nums(n);
    input(nums);
    cout << findMaxConsecutiveOnes(nums)<< endl;
    //display(nums);
    return 0;
}

File: revision/max_consecutive_ones.py
from typing import List
def maxConsecutiveOnes(nums:List[int])->int:
    check, result = 0, 0 
    for i in nums:
        if i == 0:
            check = 0
        else:
            check+=1
            result = max(result, check)
    return result
    #One line Solution:
    return len(max(''.join(map(str,nums)).split('0'),key = len))
nums = list(map(int, input().split()))
print(maxConsecutiveOnes(nums))

